packages/near-fast-auth-signer/src/components/SignMultichain/SignMultichain.tsx

```diff
@@ -0,0 +1,226 @@
+import { borshDeserialize } from 'borsher';
+import React, {
+  useEffect, useState, useCallback, useRef
+} from 'react';
+
+import { derivationPathSchema } from './schema';
+import { DerivationPathDeserialized, MultichainIframeMessage } from './types';
+import {
+  validateMessage,
+  getSignedDelegateBase64,
+  getTokenAndTotalPrice,
+  multichainAssetToNetworkName,
+  shortenAddress,
+  getGasFee
+} from './utils';
+import { getAuthState } from '../../hooks/useAuthState';
+import useFirebaseUser from '../../hooks/useFirebaseUser';
+import useIframeDialogConfig from '../../hooks/useIframeDialogConfig';
+import InternetSvg from '../../Images/Internet';
+import ModalIconSvg from '../../Images/ModalIcon';
+import { Button, CloseButton } from '../../lib/Button';
+import { ModalSignWrapper } from '../Sign/Sign.styles';
+import TableContent from '../TableContent/TableContent';
+import { TableRow } from '../TableRow/TableRow';
+
+// TODO: Delete after demo
+const sampleMessageForEthereum: MultichainIframeMessage = {
+  nearAccountId:    'osman.testnet',
+  chainId:          BigInt('5'),
+  derivationPath:   'AwAAAEVUSAAXAAAAdGVzdC1tY2hhaW4tZTJlLnRlc3RuZXQ=',
+  to:               '0x47bF16C0e80aacFf796E621AdFacbFaaf73a94A4',
+  value:            BigInt('10000000000000000')
+};
+
+function SignMultichain() {
+  const { loading: firebaseUserLoading, user: firebaseUser } = useFirebaseUser();
+  const signTransactionRef = useRef(null);
+  const [amountInfo, setAmountInfo] = useState<{ price: string | number, tokenAmount: string | number }>({ price: '...', tokenAmount: 0 });
+  const [message, setMessage] = useState<MultichainIframeMessage>(null);
+  const [inFlight, setInFlight] = useState(false);
+  const [error, setError] = useState(null);
+  const [deserializedDerivationPath, setDeserializedDerivationPath] = useState<DerivationPathDeserialized>(null);
+
+  // Send form height to modal if in iframe
+  useIframeDialogConfig({
+    element: signTransactionRef.current,
+    onClose: () => window.parent.postMessage({ signedDelegates: '', error:  'User cancelled action' }, '*')
+  });
+
+  const onError = (text: string) => {
+    window.parent.postMessage({ signedDelegates: '', text }, '*');
```

Comment:
this is a different payload than the error on line 47. Can we standardize and also would be useful if you documented the possible requests and responses in the PR or on a README on the folder

Reply:
I will try to make it similar withn SignMultichain component, but convention is already diverged on everywhere else in this repo. (eg. Existing Sign component)

Reply:
Created issue for handling above issue across whole repo https://github.com/near/fast-auth-signer/issues/183
Should be handled separately later